### PR TITLE
Add client credentials grant type to OAuth2 - N8N-3836

### PR DESCRIPTION
**Pull request by RicardoE105**
https://community.n8n.io/t/add-client-credentials-grant-type-feature-for-oauth2/5618

### CODE REVIEW

File: packages/workflow/src/Interfaces.ts [hunk 1]
```diff
@@ -1535,3 +1535,19 @@ export interface IConnectedNode {
 	indicies: number[];
 	depth: number;
 }
+
+export enum OAuth2GranType {
```
Comment (pemontto) [inline packages/workflow/src/Interfaces.ts:1539]: Minor type, this should probably be `OAuth2GrantType`

Comment (RicardoE105) [reply packages/workflow/src/Interfaces.ts:1539]: Ahh yes, good catch.

**Comment by janober** [comment]
Thanks a lot got merged.

**Comment by janober** [comment]
Got released with `n8n@0.182.0`

**Pull request merged**